### PR TITLE
Format astro files in examples

### LINKED ISSUE #3855: 🐛 BUG: Very inconsistent formatting in default example projects
### What version of `astro` are you using?

v1.0.0-beta.64

### Are you using an SSR adapter? If so, which one?

No

### What package manager are you using?

npm

### What operating system are you using?

Windows

### Describe the Bug

Not sure if this is the right place for this, but it felt better than making an RFC (happy to move it somewhere else though)

All of the example projects' formatting are very inconsistent, which is not a great user experience. The first file I opened look like this, just to give an example from the `Just the basics` template (linked repro).
|  image |  image |  image |
| --- | --- | --- |
| Inconsistent indentation (tabs vs spaces) | Mixed between ending self-closing tags with `/` or not | Mix between semicolons and commas, and inconsistent spacing & typing (two different components) |

I got formatting working using the wonderful [prettier-plugin-astro](https://github.com/withastro/prettier-plugin-astro), but users shouldn't have to deal with setting this up, the defaults should be sane even if they don't want prettier. I know the prettier plugin is in the early stages (and that's probably why `.astro` files are exlcuded in the root `.prettierignore`), but I have two questions:

# TLDR
Formatting non-existent in examples.

1. When can we expect to automatically format the entire monorepo (with all `.astro` files) with prettier?
2. Is there any interest to get just the examples decently formatted before that happens?

I have gotten the formatter running in the astro repo locally so I could submit a PR with only the format changes (without hooking up the plugin) if there's any interest. Well, I could PR hooking up the plugin too 😄 

I'll CC @Princesseuh on this one!

### Link to Minimal Reproducible Example

https://stackblitz.com/github/withastro/astro/tree/latest/examples/basics?file=src/layouts/Layout.astro

### Participation

- [X] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by MarcusOtter**
## Changes

This PR runs the [prettier-plugin-astro](https://github.com/withastro/prettier-plugin-astro) on the example projects to close #3855.

I did this by first installing the plugin in the repo, then running `npx prettier --write ./examples/**/*.astro`, then reverting the setup commit (since we do not want to add it yet, as discussed in the issue).

## Testing

I have not had time to test all the examples, would very much appreciate if someone could help me with that! It should only be a formatting change but maybe we'll find some edgecase where it breaks behavior? EDIT: I guess the CI should take care of testing that anyways.

## Docs

No docs change needed

### CODE REVIEW

File: examples/portfolio/src/pages/404.astro [hunk 1]
```diff
@@ -4,16 +4,16 @@ import Footer from '../components/Footer/index.jsx';
...
-		<nav />
-		<div class="wrapper mt4 mb4">
+		<nav></nav>
+		<div class='wrapper mt4 mb4'>
 			<h1>Page Not Found</h1>
 			<p>Not found</p>
 		</div>
-		<footer />
+		<footer></footer>
 	</body>
```
Comment (Princesseuh) [inline examples/portfolio/src/pages/404.astro:18]: This is unrelated to the formatting changes, but isn't this file supposed to use the imported components? 😅

**Comment by Princesseuh** [comment]
This looks good! Just three things:
- HTML quotes got transformed to single quotes, which is valid as in that it works in Astro, but is not recommended. Caused by https://github.com/withastro/prettier-plugin-astro/issues/174
- Slots should be self-closing but are not due to https://github.com/withastro/prettier-plugin-astro/issues/228
- A style tag was moved inside the body due this compiler issue https://github.com/withastro/compiler/issues/318

What I suggest we do is: 
- Rerun it on the examples with `singleQuotes` set to `false`, just so we get proper quotes at least
- Close slots manually
- Move the style tag manually

That way we can at least get this PR merged without needing to fix all the issues (that we'll definitely fix in the future, of course!)

**Review by Princesseuh — APPROVED** [review]
Looks good to me, there was a few changes I wasn't sure of inside the `portfolio` example but testing locally, there seems to be no issues!

**Comment by MarcusOtter** [comment]
Thanks again! I was going to go above and beyond to use `'` for inside code fences, I even made this beautiful regex for it to replace them all at once, but VSCode didn't support variable length lookbehinds 😢

If anyone is interested in doing it manually, this regex works in VSCode to select all codefences with double quotes in them: `---(.|\n)*"+(.|\n)*---`, and this is the regex that would only select the double quotes, but wasn't supported in VSCode: `(?<=---(.|\n)*)"+(?=(.|\n)*---)`

| Regextester | VSCode |
| --- | --- |
| bild | bild |

**Pull request merged**